src/main/java/seedu/address/model/article/Link.java

```diff
@@ -0,0 +1,65 @@
+package seedu.address.model.article;
+
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents an Article's link in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidLink(String)}
+ */
+public class Link implements Comparable<Link> {
```

Comment:
There is no need to implement the comparable interface to make `Link` objects comparable to one another since we do not have features that require sorting by link 'ordering'. It does not make sense as well to be doing anything of that sort.